mne/channels/layout.py

```diff
@@ -882,13 +888,32 @@ def read_montage(kind, ch_names=None, path=None, scale=True):
         dtype = [('type', 'S8'), ('name', 'S8'),
                  ('x', 'f8'), ('y', 'f8'), ('z', 'f8')]
         data = np.loadtxt(fname, dtype=dtype)
-        pos_ = data[data['type'].astype(np.str) == 'eeg']
-        pos = np.vstack((pos_['x'], pos_['y'], pos_['z'])).T
-        ch_names_ = pos_['name'].astype(np.str)
+        pos = np.vstack((data['x'], data['y'], data['z'])).T
+        ch_names_ = data['name'].astype(np.str)
     else:
         raise ValueError('Currently the "%s" template is not supported.' %
                          kind)
     selection = np.arange(len(pos))
+
+    if transform is True:
+        pos = apply_trans(als_ras_trans_mm, pos)
```

Comment:
the als to ras transform was in your original read_montage_polhemus function, it was just a transfer. 

for the als_to_ras_transform, are you saying that it is unnecessary step? I know that neuromag-trans sets up the coord system with the right-hand head-centered axis, but I thought it needed to be first transformed from ALS space to RAS space. that is what we do with all the other data and what was done before.

to avoid the scaling issue, we could make transform take a string in addition. if it is specified 'polhemus', we do both transforms of rescaling and reorientation.